src/YoutubeAtomSticky.tsx

```diff
@@ -113,8 +126,11 @@ export const YoutubeAtomSticky = ({
     isMainMedia,
     children,
 }: Props): JSX.Element => {
+    const isMobile = detectMobile();
```

Comment:
Do you think it's possible and beneficial to  make `isMobile` a ref since it should be a static value and doesn't need to change? If I'm not wrong, that way we'd call `detectMobile` only once when the component is created. 
`const isMobile = useRef(detectMobile());`

Reply:
That's an interesting one! Had a long chat with @mxdvl about the advantages and disadvantages of using `refs` before realising the blindingly obvious thing to do here is to just move the function call outside of the component 🙂 

For added context though the React docs do suggest caution when using refs: https://reactjs.org/docs/refs-and-the-dom.html

When we were working on and around the `YoutubeAtomPlayer` there were a number of reasons to use refs, but it's good to remember that React is meant to be reactive - if we use refs everywhere while it might save re-renders it also side-steps some deliberate design features in the way React is written.

Reply:
Nice explanation, and thanks for sharing the link 👍

Reply:
> Do you think it's possible and beneficial to make isMobile a ref since it should be a static value and doesn't need to change?

Usually the determining factor for using a ref is that it is not related to the render state but you still want to persist or mutate it across renders.

So just to expand on this particular case - iiuc `isMobile` is connected to the render state so shouldn't be a ref.

But refs do have legitimate use cases so it's perfectly fine to use them where they make sense.

> For added context though the React docs do suggest caution when using refs: https://reactjs.org/docs/refs-and-the-dom.html

Just icymi I think the linked docs are outdated as it discusses issues with pre-hooks refs and mainly as handles to dom elements. The React docs are in dire need of an update!

A more up to date reference is here:
https://reactjs.org/docs/hooks-reference.html#useref

Also the (vastly improved) beta of the new React docs are here:
https://beta.reactjs.org/apis/useref

Reply:
Thanks @arelra -good spot! Also those new beta docs really are infinitely better.